src/NUnitEngine/nunit.engine.api/TestPackage.cs

```diff
@@ -26,39 +26,40 @@ namespace NUnit.Engine
     public class TestPackage
     {
         /// <summary>
-        /// Construct a named TestPackage, specifying a file path for
-        /// the assembly or project to be used.
+        /// Construct an anonymous TestPackage that wraps test files.
+        /// Each file is added to the package as a subpackage.
         /// </summary>
-        /// <param name="filePath">The file path.</param>
-        public TestPackage(string filePath)
+        /// <param name="testFiles">Names of all the test files</param>
+        /// <remarks>
+        /// Semantically equivalent to the IList constructor.
+        /// </remarks>
+        public TestPackage(params string[] testFiles)
         {
-            ID = GetNextID();
-
-            if (filePath != null)
-            {
-                FullName = Path.GetFullPath(filePath);
-                Settings = new Dictionary<string,object>();
-                SubPackages = new List<TestPackage>();
-            }
+            InitializeSubPackages(testFiles);
         }
 
```

Comment:
No, we didn't because it would never have been used for a single string argument - the named constructor would have been used. I added it primarily for ease of use in tests, but then I thought users like ease of use as well. We could have it as the only constructor, but would have to change two or three places where we use the IList constructor. I can go any way you like on this.

Reply:
Personally, I prefer the clean API with a single constructor, over the convenience of both. 

It's not one I feel incredibly strongly on though - if you particularly disagree here. 🙂

Reply:
I don't have a strong preference in the abstract, but when I tried out each of the two alternatives separately, I was forced to write ugly code in one place or another. Runners and project loaders would usually have some kind of dynamic list and would be forced to construct an array from it. OTOH the 39 uses of the params constructor would have to add a new [] construct to their code.